sklearn/calibration.py

```diff
@@ -854,6 +854,20 @@ def _sigmoid_calibration(predictions, y, sample_weight=None):
 
     F = predictions  # F follows Platt's notations
 
+    # Approximated using logit(np.finfo(np.float64).eps) which is about -36
+    threshold = 30
```

Comment:
Let's put this as kwarg of `_sigmoid_calibration` named `max_abs_prediction_threshold` and write a unit test to check that for problems with only small prediction values (e.g. between -1 and 1, setting the threshold to 0.1 or 10 does not change anything but that for problem with very large prediction values, it helps fix the numerical stability problem.

Reply:
This should be a new and separate test right?

Reply:
Yes. We can keep the end-to-end non regression test as such. The goal of the unit test is to check that the specific choice of the threshold is not important.